gcc/testsuite/rust/execute/torture/builtin_macros3.rs

```diff
@@ -0,0 +1,24 @@
+// { dg-output "14\n24\n" }
+macro_rules! column {
+    () => {{}};
+}
+
+extern "C" {
+    fn printf(fmt: *const i8, ...);
+}
+
+fn print(s: u32) {
+    printf("%u\n\0" as *const str as *const i8, s);
+}
+
+fn main() -> i32 {
+    let c0 = column!();
```

Comment:
I did some testing on this and I think the issues must be relating to transcribing of macros. If you change this line to:

```
let c0:u32 = column!();
```

You will hit the issue the location info here is not the best but the error is:

```
test.rs:15:5: error: expected [u32] got [()]
   15 |     let c0: u32 = column!();
      |     ^       ~
```

The transcribed version of the code is actually

```
let c0:u32 = {}
```

This is why we get the errors we see here, I think @CohenArthur will have a better idea what to look at next